internal/plan/plan.go

```diff
@@ -258,6 +245,93 @@ const (
 	ActionIgnore   ServiceAction = "ignore"
 )
 
+type LogTargets struct {
+	Targets []string
+	Keyword string
+	Replace bool
+}
+
+const (
+	// Valid keywords for LogTargets
+	DefaultLogTargets = "default"
+	AllLogTargets     = "all"
+	NoLogTargets      = "none"
+
+	// Tag used to replace value (rather than append)
+	logTargetsReplaceTag = "!replace"
+)
+
+func (t *LogTargets) UnmarshalYAML(value *yaml.Node) error {
+	// Check for !replace tag
+	t.Replace = (value.Tag == logTargetsReplaceTag)
+
+	if value.Kind == yaml.ScalarNode {
+		switch value.Value {
+		case DefaultLogTargets, AllLogTargets, NoLogTargets:
+			t.Keyword = value.Value
+			return nil
+		default:
+			return fmt.Errorf("invalid value %q for LogTargets", value.Value)
+		}
+	}
+
+	// unmarshal to []string
+	return value.Decode(&t.Targets)
+}
+
+// Copy returns a deep copy of this LogTargets struct.
+func (t *LogTargets) Copy() LogTargets {
```

Comment:
I think this should probably return a `*LogTargets` like the other `Copy` methods.

Reply:
yep, which means we should be using `*LogTargets` instead of `LogTargets` everywhere